app/src/main/java/com/example/paritosh/copycat/MainActivity.kt

```diff
@@ -1,12 +1,49 @@
 package com.example.paritosh.copycat
 
-import android.support.v7.app.AppCompatActivity
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
 import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.widget.Switch
 
 class MainActivity : AppCompatActivity() {
 
+    private val settingSharedPreferences = "settingSharedPreferences"
+    private val permissionBoolean = "settingSharedPreferences"
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        val switch = findViewById<Switch>(R.id.permissionSwitch)
+        val sharedPreferences = getSharedPreferences(settingSharedPreferences, Context.MODE_PRIVATE)
+        val editor: SharedPreferences.Editor = sharedPreferences.edit()
+
+
+
+        switch.isChecked = sharedPreferences.getBoolean(permissionBoolean, false)
+        switch.setOnCheckedChangeListener { buttonView, isChecked ->
+            if (isChecked) {
+                modifyingPreferences(editor, isChecked)
+                settingIntent(CopyService.ACTION_START_FOREGROUND_SERVICE)
+            } else {
+                modifyingPreferences(editor, isChecked)
+                settingIntent(CopyService.ACTION_STOP_FOREGROUND_SERVICE)
+            }
+        }
+    }
+
+    private fun settingIntent(setServiceState: String) {
```

Comment:
A method name should never be like `settingIntent`. It should either be like a command or request. `postActionToCatService` seems appropriate here. Also, the name of the parameter should be `action` and not `setServiceState` which seems like a command. Parameters should be named like a piece of data as they are inputs to methods.